assembly/src/json.ts

```diff
@@ -725,3 +905,8 @@ function parseDate(dateTimeString: string): Date {
   // is globally aliased to wasi_Date (or some other superclass).
   return new Date(d.getTime());
 }
+
+// @ts-ignore: Decorator
+@inline function isMap<T>(): bool {
+  return nameof<T>().startsWith("Map<");
```

Comment:
@JairusSW - Happy to change this if you can think of a better way.  Thanks.

Reply:
I'd use some functionality like this to do that at compile time
```js
// @ts-ignore: Decorator
@inline function isMap<T>(): boolean {
    let type = changetype<T>(0);
    return type instanceof Map;
}
```
Its pretty wack, but it works well

Reply:
Perfect.  Thanks!